### PR TITLE
SIP-118  Conditionally skipping fee reclamation

### PR DESCRIPTION
**Pull request by i-stam**
When opening a pull request to submit a new SIP, please use the suggested template: https://github.com/Synthetixio/SIPs/blob/master/sip-X.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing WIP PRs.
 - The build passes.
 - Your Github username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: sips/sip-config-waiting-period.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+author:
+discussions-to:
+created: 2020-02-25
+---
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Simply describe the outcome the proposed changes intends to achieve. This should be non-technical and accessible to a casual community member.-->
+
+Allow waiting period to be set to zero and avoid unecessary fee reclamation functionality.
```
Comment (justinjmoses) [inline sips/sip-config-waiting-period.md:14]: `s/unecessary/unnecessary`

File: sips/sip-config-waiting-period.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Abstract
+
+<!--A short (~200 word) description of the proposed change, the abstract should clearly describe the proposed change. This is what *will* be done if the SIP is implemented, not *why* it should be done or *how* it will be done. If the SIP proposes deploying a new contract, write, "we propose to deploy a new contract that will do x".-->
+
+[SIP-114](./sip-114.md) will enable synth exchanging on L2. Given that transaction (and hence oracle) frontrunning is not possible on Optimisic Ethereum (OE), there is no need for storing exchanges and settling them after a certain amount of time has elapsed. This SIP proposes to set the waiting period to 0 when applicable, as is the case with Optimism.
```
Comment (justinjmoses) [inline sips/sip-config-waiting-period.md:20]: frontrunning is much less likely on L2 but technically could still be possible.

File: sips/sip-config-waiting-period.md (hunk 1 above)
Comment (justinjmoses) [reply sips/sip-config-waiting-period.md:14]: I would say "bypass fee reclamation requirement when the waiting period is set to 0"

File: sips/sip-config-waiting-period.md (hunk 2 above)
Comment (justinjmoses) [reply sips/sip-config-waiting-period.md:20]: And setting it to 0 would technically be an SCCP - this is a SIP that contains the code changes to skip fee reclamation if and when the waiting period is 0

File: sips/sip-config-waiting-period.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+In Phase 1 of the [transition to Optimistic Ethereum](https://blog.synthetix.io/the-optimistic-ethereum-transition), basic synth functionality is going to be enabled, already proposed in [SIP-114](./sip-114.md). Every time a user exchnages a synth for another, the trade is stored temporarily and can be settled after a certain waiting period (`waitingPeriodSecs` in `SystemSettings`) has elapsed, currently set to 6 minutes. This is mainly done to address oracle front running i.e. monitoring the mempool for oracle price updates and making a favourable trade before the price gets updated. After the aforementioend waiting period has expired, the exchange can be settle on the fair price and the user gets either a rebate or a reclaim. This whole procedure is not needed on OE, thus the waiting period can be set to 0 and skip part of the overhead imposed by settlement.
```
Comment (justinjmoses) [inline sips/sip-config-waiting-period.md:26]: `s/exchnges/exchanges`

Comment (justinjmoses) [reply sips/sip-config-waiting-period.md:26]: I don't think you need to rehash what fee reclamation is- you can link SIP-37 instead.

File: sips/sip-config-waiting-period.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+4. Test Cases
+5. Configurable Values
+-->
+
+### Overview
+
+<!--  -->
+<!--This is a high level overview of *how* the SIP will solve the problem. The overview should clearly describe how the new feature will be implemented.-->
+
+No new contracts were developed for this SIP, only contract functionality was slightly altered. The waiting period could be set to 0 when deploying an OVM instance and no rebate/reclaim has to take place. This can done by specifying a WAITING_PERIOD_SECS entry in the `params.json` file of the corresponding deployment folder. In the `Exchanger` contract, an extra check is added to prevent storing exchanges and subsequently having to settle them, reducing in this way both the computational and storing overhead and saving gas.
```
Comment (justinjmoses) [inline sips/sip-config-waiting-period.md:43]: Please use future tense.

**Review by justinjmoses — CHANGES_REQUESTED** [review]
A few comments inline

File: sips/sip-config-waiting-period.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+---
+sip: 118
+title: Conditionally skipping fee reclamation
+status: Proposed
+author:
```
Comment (ajsantander) [inline sips/sip-config-waiting-period.md:5]: Section is incomplete

File: sips/sip-config-waiting-period.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                exchangeFeeRate
+            );
+        }
+```
+
+### Test Cases
+
+<!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
+
+TBD
```
Comment (ajsantander) [inline sips/sip-config-waiting-period.md:76]: Incomplete?

**Review by ajsantander — CHANGES_REQUESTED** [review]
Looks good. Suggested some changes in https://github.com/Synthetixio/SIPs/pull/406

File: sips/sip-118.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+### Test Cases
+
+<!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
+
+TBD
+
+### Configurable Values (Via SCCP)
+
+<!--Please list all values configurable via SCCP under this implementation.-->
```
Comment (justinjmoses) [inline sips/sip-118.md:80]: perhaps just add a point here on the existing `waitingPeriodSecs`

File: sips/sip-118.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
+---
+sip: 118
+title: Conditionally skipping fee reclamation
+status: Proposed
+author: Yannis Stamelakos (@i-stam), Justin J Moses (@justinjmoses), Alejandro Santander (@ajsantander)
+discussions-to:
```
Comment (justinjmoses) [inline sips/sip-118.md:6]: Just a new discussion thread required

**Review by justinjmoses — APPROVED** [review]

**Pull request merged**